src/driver/postgres/PostgresQueryRunner.ts

```diff
@@ -747,17 +748,19 @@ export class PostgresQueryRunner extends BaseQueryRunner implements QueryRunner
                     downQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${newColumn.name}" DROP DEFAULT`));
                 }
 
-                // remove old ENUM
-                upQueries.push(this.dropEnumTypeSql(table, newColumn, oldEnumName));
-                downQueries.push(this.createEnumTypeSql(table, oldColumn, oldEnumName));
+                // up: remove old renamed enum type
+                // down: rename new enum type and create old enum type
+                upQueries.push(this.dropEnumTypeSql(table, newColumn, prevEnumNameWithOldSuffixAndSchema));
+                downQueries.push(this.createEnumTypeSql(table, oldColumn));
+                downQueries.push(new Query(`ALTER TYPE ${newEnumNameWithSchema} RENAME TO ${newEnumNameWithOldSuffix}`));
             }
 
             if (oldColumn.isNullable !== newColumn.isNullable) {
                 if (newColumn.isNullable) {
-                    upQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${oldColumn.name}" DROP NOT NULL`));
+                    upQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${newColumn.name}" DROP NOT NULL`));
```

Comment:
Switched from old column to new column.  Intended?

Reply:
At this point i believe technically `oldColumn` shouldn't exist, it would have already been renamed. but that said, i dont believe that the postgres generated migrations actually ever rename a column (always seems to drop and recreate) so it may be a moot point.